File: Backend/parse_restaurants.js
const url = 'https://map-places.p.rapidapi.com/queryautocomplete/json?input=pizza%20near%20Sydney&radius=50000';
const options = {
	method: 'GET',
	headers: {
		'X-RapidAPI-Key': '080f4ef5cfmsh1bc3dcd3a9de26ap1dd994jsnebba317159df',
		'X-RapidAPI-Host': 'map-places.p.rapidapi.com'
	}
};

try {
	const response = await fetch(url, options);
	const result = await response.text();
	console.log(result);
} catch (error) {
	console.error(error);
}

File: Backend/parse_restaurants.py
import requests
import streamlit as st

zipcode = 10014
url = f"https://restaurants-near-me-usa.p.rapidapi.com/restaurants/location/zipcode/{zipcode}/0"

headers = {
	"X-RapidAPI-Key": "080f4ef5cfmsh1bc3dcd3a9de26ap1dd994jsnebba317159df",
	"X-RapidAPI-Host": "restaurants-near-me-usa.p.rapidapi.com"
}

response = requests.get(url, headers=headers)
json_response = response.json()

open_restaurants = list()

for restaurant in json_response["restaurants"]:
    cuisines = restaurant["cuisineType"].split(",")
    open_restaurants.append({
        "name":restaurant["restaurantName"],
        "address":restaurant["address"],
        "phone":restaurant["phone"],
        "state":restaurant["stateName"],
        "city":restaurant["cityName"],
        "cuisines":cuisines
    })

print(open_restaurants)